### ISSUE TITLE
Failed to create connection for connection id '/providers/Microsoft.PowerApps/apis/shared_powerplatformadminv2/... Connection creation/edit of 'shared_powerplatformadminv2' has been blocked by Data Loss Prevention (DLP) policy 

### ISSUE THREAD

**Issue by RegassaKedir** [microsoft/coe-starter-kit#9099]
### Does this bug already exist in our backlog?

- [x] I have checked and confirm this is a new bug.

### Describe the issue

During upgrade we are getting an an error "Failed to create connection for connection id '/providers/Microsoft.PowerApps/apis/shared_powerplatformadminv2/connections/shared-powerplatform-ddc17df0-4d92-4980-884f-c49c0125ec80'. Connection creation/edit of 'shared_powerplatformadminv2' has been blocked by Data Loss Prevention (DLP) policy 'Global Org-wide Policy'."

### Expected Behavior

Expected to run the update without an issue. We couldn’t update the solution due to the error.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.32

### What app or flow are you having the issue with?

Upgrading COE Kit

### What method are you using to get inventory and telemetry?

Cloud flows

**Comment by Jenefer-Monroe** [comment]
Hello this is a new connector used in the kit, you will need to upgrade the DLP policy for the envt. 
Please see [Validate data loss prevention (DLP) policies](https://learn.microsoft.com/power-platform/guidance/coe/setup#validate-data-loss-prevention-dlp-policies)

**Comment by RegassaKedir** [comment]
Jennifer, thank you. I was able to adjust the DLP policy and resolve the issue. However, after doing that, I encountered another problem. The solution failed to update and couldn’t remove and replace the old solution, resulting in the following error. Can you please help?

"Information
Error code
8004F01D
Exception message
Solution dependencies exist, cannot uninstall. DependencyCount : 1 RequiredComponentObject details:...
See more

Total operation time
4 sec
Patch
Yes
Managed
Yes
Overwrite customizations
No
Activity id"

**Comment by Jenefer-Monroe** [comment]
there have been some issues lately with the upgrade code. Can you get me the rest of the error message?

**Comment by RegassaKedir** [comment]
Error code
[8004F01D](http://go.microsoft.com/fwlink/?LinkID=398563&client=Solutions&Error=8004F01D&de=online)
Exception message
Solution dependencies exist, cannot uninstall. DependencyCount : 1 RequiredComponentObject details: Type: SavedQuery, ObjectName: Abandoned PowerApps, Id: f9d327af-b6b4-e911-a85b-000d3a372932, ParentObjectName: PowerApps App, ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: AppModule, ObjectName: epacoe_PowerPlatformUserView, Id: eeb4bcb7-ea12-ef11-9f8a-001dd8306c96, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published
See less

Total operation time
4 sec
Patch
Yes
Managed
Yes
Overwrite customizations
No
Activity id
5bd43a56-cea2-4376-915e-f400af5a7976

**Comment by RegassaKedir** [comment]
Hi Jenefer - the above error is from the history tab, and I also included the below error that appears as a banner when the failure happens

"Solution "Center of Excellence - Core Components" failed to import: Solution Upgrade action failed after import as holding. InnerException is: Solution dependencies exist, cannot uninstall. DependencyCount : 1 RequiredComponentObject details: Type: SavedQuery, ObjectName: Abandoned PowerApps, Id: f9d327af-b6b4-e911-a85b-000d3a372932, ParentObjectName: PowerApps App, ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: AppModule, ObjectName: epacoe_PowerPlatformUserView, Id: eeb4bcb7-ea12-ef11-9f8a-001dd8306c96, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published ."

**Comment by Jenefer-Monroe** [comment]
Hello. A few other users have hit this and we were able to find a workaround. See https://github.com/microsoft/coe-starter-kit/issues/8682
**The change:** 
We removed a view called **_Abandoned PowerApps_** from the kit as it was found to be a confusing duplicate. 
We did remove it from the MDA's at that time as well of course or it would not have let me delete it. 
**The issue:**
For some reason the product's upgrade code is choking on this for some users. 
I think its because its reviewing all the views and seeing that one has a dependency before checking to see if that dependency had been removed from the MDA. 
**The workaround**
This worked for the other person, you can try to remove our dependency on it manually and see if that fixes
1. Remove the _upgrade core solution if it exists
2. Go to Apps > find the Power Platform Admin View app > Edit
![image](https://github.com/user-attachments/assets/7a36f8f4-6b25-4f6b-8d24-a59ce01040be)
3. Remove the problematic view and publish
![image](https://github.com/user-attachments/assets/fe365cfb-12c3-4dff-bde5-1864b1371381)
4. Try your upgrade again.

**Comment by RegassaKedir** [comment]
![Image](https://github.com/user-attachments/assets/21115458-d07d-4816-8b32-82eccb18ba78)
I followed the procedure removed the "Abandoned Power App" view. Tried to update again and another error attached. 

Here is the failure from the history tab
"Error code
[8004F01D](http://go.microsoft.com/fwlink/?LinkID=398563&client=Solutions&Error=8004F01D&de=online)
Exception message
Solution dependencies exist, cannot uninstall. DependencyCount : 1 RequiredComponentObject details: Type: SavedQuery, ObjectName: Abandoned PowerApps, Id: f9d327af-b6b4-e911-a85b-000d3a372932, ParentObjectName: PowerApps App, ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: AppModule, ObjectName: epacoe_PowerPlatformUserView, Id: eeb4bcb7-ea12-ef11-9f8a-001dd8306c96, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published
See less

Total operation time
4 sec
Patch
Yes
Managed
Yes
Overwrite customizations
No
Activity id
56d88c0f-7516-4726-ae75-7d40945ec536"

**Comment by DarkestOfNights** [comment]
@RegassaKedir - I know of this problem personally. The problem is the copy of the Power Platform Admin view that was made for non-admin users a while back - "Power Platform User View". It hadn't interfered with updates previously, but must with recent updates to the CoE. If you remove that app, you should be able to perform the upgrade. - TH

**Comment by Jenefer-Monroe** [comment]
As @DarkestOfNights indicates it appears you have a custom app in your tenant that still has the dependency. In the error message it shows its name, so you will need to find it and either remove the dependency or delete the app. 

![Image](https://github.com/user-attachments/assets/c2151b75-f583-4330-a236-9ed0aae89762)

Thanks for your help!

**Comment by RegassaKedir** [comment]
Jenefer - We deleted the app ‘Power Platform Admin View’. We tried to delete the CoE core component upgrades to reinstall it again and encountered another issue.

"Error code
[8004F01D](http://go.microsoft.com/fwlink/?LinkID=398563&client=Solutions&Error=8004F01D&de=online)
Exception message
Solution dependencies exist, cannot uninstall. DependencyCount : 6 RequiredComponentObject details: Type: SavedQuery, ObjectName: Shared With Guests - Owner, Id: b722f4c3-a52c-ef11-840a-000d3a352a2a, ParentObjectName: PowerApps App, ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: AppModule, ObjectName: admin_PowerPlatformAdminView, Id: f1be0217-5fa5-ec11-b400-001dd804f8b1, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published RequiredComponentObject details: Type: SavedQuery, ObjectName: Shared With Guests - Owner, Id: cfa20cb6-a52c-ef11-840a-00224803bae7, ParentObjectName: Flow, ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: AppModule, ObjectName: admin_PowerPlatformAdminView, Id: f1be0217-5fa5-ec11-b400-001dd804f8b1, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published RequiredComponentObject details: Type: SavedQuery, ObjectName: Shared With Guest - Editor, Id: 876fb7b2-ef2c-ef11-840b-0022480c0873, ParentObjectName: PowerApps App, ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: AppModule, ObjectName: admin_PowerPlatformAdminView, Id: f1be0217-5fa5-ec11-b400-001dd804f8b1, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published RequiredComponentObject details: Type: SavedQuery, ObjectName: Shared With View, Id: dd84538d-a02a-ef11-840a-6045bd07060c, ParentObjectName: Flow, ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: AppModule, ObjectName: admin_PowerPlatformAdminView, Id: f1be0217-5fa5-ec11-b400-001dd804f8b1, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published RequiredComponentObject details: Type: SavedQuery, ObjectName: Shared With Guests, Id: f5812025-a12a-ef11-840a-6045bd07060c, ParentObjectName: Flow, ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: AppModule, ObjectName: admin_PowerPlatformAdminView, Id: f1be0217-5fa5-ec11-b400-001dd804f8b1, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published RequiredComponentObject details: Type: SavedQuery, ObjectName: Shared With Guest - Editor, Id: 9c3e32be-ef2c-ef11-840a-6045bd07060c, ParentObjectName: Flow, ParentId: 00000000-0000-0000-0000-000000000000, DependentComponentObject details: Type: AppModule, ObjectName: admin_PowerPlatformAdminView, Id: f1be0217-5fa5-ec11-b400-001dd804f8b1, ParentObjectName: , ParentId: 00000000-0000-0000-0000-000000000000 DependencyType: Published
See less

Total operation time
7 sec
Patch
Yes
Managed
Yes
Overwrite customizations
No
Activity id
7ce4a387-e35d-494d-930e-7ea8007b1632"

**Comment by Jenefer-Monroe** [comment]
You may benefit from trying a classic overwrite

Classic overwrite upgrade
### How to do an overwrite upgrade in the classic UX:
 
Switch to classic
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/690cafcb-ade8-4571-81d5-df70f97f6266)

Import
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/433cd2d8-7e2a-4d35-a86e-cf569b7777f2)

Select Core > Next
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/f232f4b5-783d-4d8c-af94-21fb6dfd91bf)

Next
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/39c84d9b-7ccf-4aff-8861-6c70ed7cb237)

Overwrite customizations > Import
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/7b23463c-2361-4135-81a8-363a96ca23d5)
****

**Comment by RegassaKedir** [comment]
Jenefer - Thank you for your help. I was able to remove the dependencies and delete the core component upgrade. I am now able to install and update the core component solution successfully.

One last question: I am trying to update the [Center of Excellence - Governance Components], but I don’t see a solution with that name in the CoE Starter Kit?![Image](https://github.com/user-attachments/assets/f6b75dc6-7ff0-4aca-bc40-6a241ecdd12e)

**Comment by Jenefer-Monroe** [comment]
Wonderful news!

The Gov components are held today in the "CenterofExcellenceAuditComponents_3_25_managed.zip" file

Please see the [Before setting up the governance components](https://learn.microsoft.com/power-platform/guidance/coe/before-setup-gov)
![Image](https://github.com/user-attachments/assets/18758a98-fbf1-436f-95fd-2ca0dee4ff14)

**Comment by RegassaKedir** [comment]
Jenefer,

We were able to successfully upgrade “CenterofExcellenceAuditComponents_3_25_managed.zip.”

However, we are now experiencing issues with the Power BI report of the Power Platform COE dashboard. The refresh has been failing for the past couple of days. Below are the error details:

"the Data source error: Expression.Error: The column 'admin_totalshared' of the table wasn't found.. admin_totalshared. . The exception was raised by the IDbCommand interface. Table: Power Apps.
Cluster URI: WABI-US-GOV-VIRGINIA-redirect.analysis.usgovcloudapi.net
Activity ID: 4d3ca2b9-8bbb-4264-9fc3-1bebc5b523c6
Request ID: 0e2892b1-2c77-b167-1079-28c0cf782f6a
Time: 2024-10-16 12:48:34Z"

**Comment by Jenefer-Monroe** [comment]
What PBIT file did you use?

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team

**Comment by RegassaKedir** [comment]
Hello Jenefer - We are still having issue with refreshing the COE Power BI report. It has been working for the last two years.
